src/extension/geofence/src/main/java/org/geoserver/geofence/GeofenceAccessManager.java

```diff
@@ -575,6 +575,8 @@ private void setRuleFilterUserOrRole(Authentication user, RuleFilter ruleFilter)
                 for (GrantedAuthority authority : user.getAuthorities()) {
                     if (config.getRoles().contains(authority.getAuthority())) {
                         role = authority.getAuthority();
+                        ruleFilter.setUser(RuleFilter.SpecialFilterType.DEFAULT);
```

Comment:
This `setUser` should be run everytime the condition `config.isUseRolesToFilter()` in the previous if block is true.
As it is now, it is only executed when a role is found.
Please move this line beside the setRole call at the end of the `if` block starting at line 572